### PR TITLE
Extend Scylla Manager E2E test with progress check

### PR DESCRIPTION
**Pull request by rzetelskik**
**Description of your changes:**
At this point the E2E for Scylla Manager integration only tests that the task was scheduled. This PR extends the test with awaiting for task's completeness.

**Which issue is resolved by this Pull Request:**
Part of the effort to improve scylla manager integration.

/kind feature
/priority important-longterm
/cc

### CODE REVIEW

File: test/e2e/set/scyllacluster/scyllamanager.go [hunk 1]
```diff
@@ -85,20 +115,32 @@ var _ = g.Describe("Scylla Manager integration", func() {
...
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(tasks.TaskListItemSlice).To(o.HaveLen(1))
 		repairTask := tasks.TaskListItemSlice[0]
-		o.Expect(repairTask.Name).To(o.Equal(sc.Spec.Repairs[0].Name))
+		o.Expect(repairTask.Name).To(o.Equal(sc.Status.Repairs[0].Name))
+		o.Expect(repairTask.ID).To(o.Equal(sc.Status.Repairs[0].ID))
 		o.Expect(repairTask.Properties.(map[string]interface{})["parallel"].(json.Number).Int64()).To(o.Equal(int64(123)))
+
+		framework.By("Waiting for repair to finish")
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(context.Context) (done bool, err error) {
```
Comment (tnozicka) [inline test/e2e/set/scyllacluster/scyllamanager.go:136]: should this be propagated to scyllacluster status? if yes and we miss the information, a todo with an issue link seems to be in order.

**Review by tnozicka — APPROVED** [review]
/approve
/asssing @zimnx

File: test/e2e/set/scyllacluster/scyllamanager.go (hunk 1 above)
Comment (rzetelskik) [reply test/e2e/set/scyllacluster/scyllamanager.go:136]: to my knowledge we don't sync the tasks post-scheduling, so it's not that we miss the status propagation, we don't have the logic to handle the progress. I'm not sure if that could work with recurring tasks though. @zimnx mind sharing your thoughts on if that's viable?

Comment (tnozicka) [reply test/e2e/set/scyllacluster/scyllamanager.go:136]: yeah, with recurring tasks we'd have to parse it and schedule queue add at that point - probably not what we want to get into - I take this back :)

File: test/e2e/set/scyllacluster/scyllamanager.go [hunk 2]
```diff
@@ -85,20 +115,32 @@ var _ = g.Describe("Scylla Manager integration", func() {
...
+			repairProgress, err := managerClient.RepairProgress(ctx, *sc.Status.ManagerID, repairTask.ID, "latest")
```
Comment (zimnx) [inline test/e2e/set/scyllacluster/scyllamanager.go:137]: add o.Expect validating whether sc.Status.ManagerID is not nil before this poll

Comment (rzetelskik) [reply test/e2e/set/scyllacluster/scyllamanager.go:140]: done

**Review by zimnx — APPROVED** [review]
/lgtm

**Comment by rzetelskik** [comment]
flake https://github.com/scylladb/scylla-operator/issues/1525#issuecomment-1979722391
/retest

**Pull request merged**